engine/evd.ml

```diff
@@ -1783,3 +1783,12 @@ let evars_of_filtered_evar_info (type a) evd (evi : a evar_info) =
        | Evar_empty -> Evar.Set.empty
        | Evar_defined b -> evars_of_term evd b)
        (evars_of_named_context evd (evar_filtered_context evi)))
+
+let drop_new_defined ~original sigma =
+  let to_keep, to_drop = Evar.Map.partition (fun ev _ ->
+      Evar.Map.mem ev original.defn_evars || Evar.Map.mem ev original.undf_evars)
+      sigma.defn_evars
+  in
+  let sigma = { sigma with defn_evars = to_keep } in
+  let dummy = { empty with defn_evars = to_drop } in
+  cmap (MiniEConstr.to_constr_nocheck dummy) sigma
```

Comment:
This is probably less efficient than we should be 
Since we're only dropping the new evars we don't need to touch the parts of the evar map which are unchanged